lib/smack/SmackModuleGenerator.cpp

```diff
@@ -67,8 +67,15 @@ void SmackModuleGenerator::generateProgram(llvm::Module& m) {
       naming.leave();
 
       // First execute static initializers, in the main procedure.
-      if (naming.get(*func) == "main" && rep.hasStaticInits())
-        proc->insert(Stmt::call(SmackRep::STATIC_INIT));
+      if (naming.get(*func) == "main") {
+        proc->insert(Stmt::call(SmackRep::INIT_FUNCS));
+        if(rep.hasStaticInits())
+          proc->insert(Stmt::call(SmackRep::STATIC_INIT));
+      } else if (naming.get(*func).substr(0, 18)  == "__SMACK_init_func_") {
+        assert(!func->doesNotReturn() && "Init functions cannot return a value");
```

Comment:
You are using a wrong API call here...
It should be:
func->getReturnType()->isVoidTy()